canvas/cli/common.py

```diff
@@ -1,16 +1,63 @@
+import json
 import sys
 
+DEFAULT_JSON = False
+DEFAULT_TABLE = False
+DEFAULT_SKIP_HEADERS = False
+
 # keys: [(items key, title, pretty title), ...]
-def cli_list(items, keys, table = False, skip_headers = False,
-        collective_name = 'items', sort_key = 'id'):
+def cli_list(items, keys, collective_name = 'items', sort_key = 'id', output_json = DEFAULT_JSON,
+        table = DEFAULT_TABLE, skip_headers = DEFAULT_SKIP_HEADERS, **kwargs):
+    items = list(sorted(items, key = lambda item: item.get(sort_key, '')))
+
+    if (output_json):
+        _cli_list_json(items = items, keys = keys)
+    elif (table):
+        _cli_list_table(items = items, keys = keys, skip_headers = skip_headers, collective_name = collective_name)
+    else:
+        _cli_list_plain(items = items, keys = keys, collective_name = collective_name)
```

Comment:
Since there are inner/private (helper) functions,
I don't think we need to pass by name.
We can just pass everything directly.
e.g. `_cli_list_json(items, keys)` instead of `_cli_list_json(items = items, keys = keys)`.